### DISCUSSION TITLE
which branch is the working branch

### DISCUSSION THREAD

**Discussion by SarveshSathishkumar** [jonatan-ivanov/teahouse#21] [Q&A] [answered]
I want to run it in my IntelliJ to run it and learn. I tried to clone 2024-springio but it says jdk mismatch for 22 jdk and 8.7 grade which I am not sure is the issue.

Please give any answer that would provide some tips for me to run it in my IDE successfully! 

Thank you!!

**Comment by jonatan-ivanov** [comment, marked as answer]
The working branch is the default branch that your git client will check out by default during clone: `main`, you don't need to do anything else.
The "conference branches" were created for those past events and not maintained anymore (they are kind of snapshots).

Even though the `2024-springio` branch should work with Java 22, I would recommend checking out `main` and using Java 23. The easiest way to import it to IntelliJ is simply running `idea .` in the project root. The easiest way of executing is `./gradlew bootRun` as the readme suggests.

**Reply by SarveshSathishkumar** [reply]
Thank you for your answer but somehow I might be having some compatibility issues with my local I guess.

Still compiles and runs but the services are "all down" in my spring boot admin but works on a colleague's system which is a MAC.
Trying to figure out how to get it fully functional.

**Reply by jonatan-ivanov** [reply]
Please check the readme for the useful urls: https://github.com/jonatan-ivanov/teahouse?tab=readme-ov-file#useful-urls. Go to the Tea UI: http://localhost:8090/steep and press the steep button, if that works Spring Boot Admin should show them up too (might need a few minutes).

If steep works, but SBA does not, try directly with the IP that SBA has and see if you get an error.
